src/main/java/com/acacia/angleddream/Main.java

```diff
@@ -132,56 +137,58 @@ public static void main(String[] args) throws IOException {
 
         if (!outputTopics.isEmpty()) {
 
-            try {
+            for(String inputTopic: inputTopics) {
+                try {
 
-                PCollection<String> inp = pipeline.apply(PubsubIO.Read.topic(options.getPubsubTopic()));
+                    PCollection<String> inp = pipeline.apply(PubsubIO.Read.topic(inputTopic));
 
```

Comment:
`PCollection<String> inp = pipeline.apply(PubsubIO.Read.topic(inputTopic));` move this outside the try and remove `for(String inputTopic: inputTopics)`.

Then create a ParDo to launch as many parallel topic reads as required which all end up in one `PCollection<String> inp`

There rest of the code will then not need to change.